units/generate_tasks.go

```diff
@@ -179,11 +179,31 @@ func (j *generateTasksJob) generate(ctx context.Context, t *task.Task) error {
 		"job":           j.ID(),
 		"version":       t.Version,
 	})
+
+	start = time.Now()
+	cycles, err := g.CyclesForNewGraph(v, p, pref)
+	if err != nil {
+		return errors.Wrap(err, "checking new dependency graph for cycles")
+	}
+	if len(cycles) > 0 {
```

Comment:
I'm wondering if this logic should just be moved into CyclesForNewGraph since this is always what we do with it.

Reply:
I moved the cycle check back into the function